src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/AvoidDeadConditionalCode.cs

```diff
@@ -151,9 +152,10 @@ public override void Initialize(AnalysisContext context)
                                         }
 
                                         var originalOperation = operationRoot.SemanticModel.GetOperation(operation.Syntax, operationBlockContext.CancellationToken);
-                                        if (originalOperation is IAssignmentOperation)
+                                        if (originalOperation is IAssignmentOperation ||
+                                            originalOperation is IVariableDeclaratorOperation)
                                         {
-                                            // Skip compiler generated IsNull operation for assignment within a using.
+                                            // Skip compiler generated IsNull operation for assignment/variable declaration within a using.
```

Comment:
I would have expected that we check that there is a parent operation being the using. I am assuming that the code was correctly handling the assignment in the using so I have simply added the variable declarator case.